pinot-core/src/main/java/org/apache/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -221,30 +223,31 @@ private boolean isLeadControllerResourceEnabled() {
   }
 
   /**
-   * Invalidates the cached controller leader value by setting the {@link ControllerLeaderLocator::_cacheControllerLeadeInvalid} flag.
+   * Invalidates the cached controller leader value by removing the existing pair from {@link ControllerLeaderLocator::_cachedValidControllerLeaderMap}.
    * This flag is always checked first by {@link ControllerLeaderLocator::getControllerLeader()} method before returning the leader. If set, leader is fetched from helix, else cached leader value is returned.
    *
    * Invalidates are not allowed more frequently than {@link ControllerLeaderLocator::MILLIS_BETWEEN_INVALIDATE} millis.
    * The cache is invalidated whenever server gets NOT_LEADER or NOT_SENT response. A NOT_LEADER response definitely needs a cache refresh. However, a NOT_SENT response could also happen for reasons other than controller not being leader.
    * Thus the frequency limiting is done to guard against frequent cache refreshes, in cases where we might be getting too many NOT_SENT responses due to some other errors.
+   * @param rawTableName raw table name.
    */
-  public synchronized void invalidateCachedControllerLeader() {
+  public synchronized void invalidateCachedControllerLeader(String rawTableName) {
```

Comment:
If the master for one table has changed, it is better to invalidate the whole cache, and fetch the EV right?

Reply:
Good point. Actually the CRUSHed algorithm used for lead controller resource guarantees least moves of the partition assignment change. It's more likely that one partition leader won't affect other partition leader.